test/provider_os_memory_config.cpp

```diff
@@ -0,0 +1,192 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include "base.hpp"
+#include "provider_os_memory_internal.h"
+#include "umf/providers/provider_os_memory.h"
+
+#include <numa.h>
+#include <numaif.h>
+#include <unistd.h>
+
+struct providerConfigTest : testing::Test {
+    umf_result_t res;
+    umf_memory_provider_handle_t provider = nullptr;
+    umf_os_memory_provider_params_t params = {
+        /* .protection = */ UMF_PROTECTION_READ | UMF_PROTECTION_WRITE,
+        /* .visibility = */ UMF_VISIBILITY_PRIVATE,
+
+        // NUMA config
+        /* .nodemask = */ NULL,
+        /* .maxnode = */ 0,
+        /* .numa_mode = */ UMF_NUMA_MODE_DEFAULT,
+        /* .numa_flags = */ 0,
+
+        // others
+        /* .traces = */ 1,
+    };
+    size_t size = 128;
+    void *ptr = nullptr;
+    char *dest = nullptr;
+
+    void SetUp() override {
+        int ret = numa_available();
+        if (ret) {
+            GTEST_SKIP() << "Test skipped, NUMA not available";
+        }
+        dest = (char *)malloc(size);
+    }
+
+    void TearDown() override {
+        if (ptr) {
+            umfMemoryProviderFree(provider, ptr, size);
+        }
+        free(dest);
+        if (provider) {
+            umfMemoryProviderDestroy(provider);
+        }
+    }
+
+    void create_provider(umf_os_memory_provider_params_t *params) {
+        res = umfMemoryProviderCreate(&UMF_OS_MEMORY_PROVIDER_OPS, params,
+                                      &provider);
+        ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+        ASSERT_NE(provider, nullptr);
+    }
+
+    void allocate_memory() {
+        res = umfMemoryProviderAlloc(provider, size, os_get_page_size(), &ptr);
+        ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+        ASSERT_NE(ptr, nullptr);
+    }
+
+    void copy_ptr_data() {
+        *dest = 'B';
+        memcpy(dest, ptr, size);
+    }
+
+    void read_memory() {
+        copy_ptr_data();
+        EXPECT_EQ(*static_cast<char *>(ptr), *dest);
+    }
+
+    void write_memory(char c) {
+        memset(ptr, c, size);
+        EXPECT_EQ(static_cast<char *>(ptr)[0], c);
+    }
+};
+
+TEST_F(providerConfigTest, protection_flag_none) {
+    // pages may not be accessed - PROT_NONE
+    params.protection = UMF_PROTECTION_NONE;
+    create_provider(&params);
+    allocate_memory();
+
+    // read failure
+    EXPECT_DEATH(copy_ptr_data(), "");
+
+    // write failure
+    EXPECT_DEATH(memset(ptr, 'A', size), "");
```

Comment:
will this line be executed after the failure for copy?

Reply:
Yes, after the expected read failure, this line will be executed with expected write failure. Read or write success will fail this test case.